src/fdc3/dotnet/DesktopAgent.Client/test/MorganStanley.ComposeUI.Fdc3.DesktopAgent.Client.Tests/Infrastructure/DesktopAgentClient.Tests.cs

```diff
@@ -425,6 +428,179 @@ public async Task Broadcast_sends_message()
                 It.IsAny<CancellationToken>()), Times.Once);
     }
 
+    [Fact]
+    public async Task GetUserChannels_returns_channels()
+    {
+        var messagingMock = new Mock<IMessaging>();
+
+        messagingMock.Setup(
+                _ => _.InvokeServiceAsync(
+                    It.IsAny<string>(),
+                    It.IsAny<string>(),
+                    It.IsAny<CancellationToken>()))
+            .Returns(new ValueTask<string?>(
+                JsonSerializer.Serialize(
+                    new GetUserChannelsResponse 
+                    { 
+                        Channels = new[] 
+                        { 
+                            new ChannelItem { Id = "1", DisplayMetadata = new DisplayMetadata { Name = "1" } }, 
+                            new ChannelItem { Id = "2", DisplayMetadata = new DisplayMetadata { Name = "2" } } 
+                        } 
+                    }, _jsonSerializerOptions )));
+
+        var desktopAgent = new DesktopAgentClient(messagingMock.Object);
+
+        var result = await desktopAgent.GetUserChannels();
+
+        result.Should().NotBeNull();
+        result.Should().HaveCount(2);
+        result.Should().BeEquivalentTo(
```

Comment:
If you check for equivalence there is not much point in checking for not null and count